modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/components/button-with-icon/ButtonWithIcon.es.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import {ClayTooltipProvider} from '@clayui/tooltip';
+import React from 'react';
+
+const ButtonWithIcon = (props) => {
```

Comment:
@gabrielpramos , esse componente é bem semelhante a um que temos no App Builder

https://github.com/liferay/liferay-portal/blob/master/modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/components/button/Button.es.js

Com excessão do TooltipProvider, tu poderia reconsiderar e utilizar o componente já existente? Ele aceita um ícone como props, além de outros parâmetros como data-tooltip-align e delay, como tem ali!

Reply:
Ok, I'll test if this component fits the mockup definition :D, Thanks!

Reply:
Uma dúvida, não conseguimos utilizar o componente ClayButton diretamente não?

Reply:
O Componente de Button já utiliza o ClayButton @interaminense